### DISCUSSION TITLE
Guests do not see results?

### DISCUSSION THREAD

**Discussion by mijnham** [microsoft-search/pnp-modern-search#2990] [Q&A]
We are using PnP search results on a communication site using MS Search and it works fine for member accounts. However, guests do get an error and we don't know why. The error is also different from the one reported on this forum before, so we think it is something else. The standard MS Search pages do give results for guests, so guest do have access to search results in theory. Could it be PnP Search is limited when it comes to guest access due to the API use? 

![image](https://github.com/microsoft-search/pnp-modern-search/assets/11948702/6577e6b8-6ba7-4838-b079-bb2cca10d31a)

**Comment by kasperbolarsen** [comment]
How does the out of the box MS Search box in the top bar behave for those users?

**Reply by mijnham** [reply]
That will give these results for guests (example with news vertical)  - hence search is accessible for guests.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/11948702/d8975d98-7154-4397-9144-03aa9de9e4d4)

**Comment by kasperbolarsen** [comment]
which Entity types are you looking for?

External Items and People is off limits for guests.

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/6099a2b9-8a02-482f-9e7f-9d435c0229c1)

**Reply by mijnham** [reply]
I have 3 'result webparts', separated with a vertical. One is external items (no guest here is ok), one is list items (=site pages) & one is drive items (= files). The latter 2 should work, no?

**Comment by kasperbolarsen** [comment]
I can recreate the scenario you are facing, and gets this exception from Graph for each search vertical when using a guest user:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/7819b194-5ca5-4679-9420-aff96499a964)

**Reply by mijnham** [reply]
Given this answer of the Graph API, my conclusion is right: this is not supported (for now)?

**Reply by wobba** [reply]
Guest user support on the Graph API does not seem to work and I have reported the same. It's not mentioned in known limitations either. You can also try to ask at Stack Overflow and tag with `microsoft-graph-api`.

**Reply by kasperbolarsen** [reply]
@wobba did you report this issue using a MS internal system or do a have a public available ticket ID we can follow?

**Reply by wobba** [reply]
Internal 🙂
